### ISSUE TITLE
Autopilot modules do not set network tags properly

### ISSUE THREAD

**Issue by TobiasLierzer** [terraform-google-modules/terraform-google-kubernetes-engine#1843]
### TL;DR

The beta-autopilot-*-modules define a `local.cluster_network_tag` which is referenced in the firewall rules. The node_pool_auto_config on the other hand only uses `var.network_tags`. Due to this mismatch, the Firewall rules do not apply to the cluster.

### Expected behavior

I'd expect that `var.network_tags` and `local.cluster_network_tag` are concatted like [here in the root/cluster.tf ](https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/blob/master/cluster.tf#L319)

### Observed behavior

The firewall rules do not apply for the cluster.

### Terraform Configuration

```hcl
module "gke" {
  source                       = "terraform-google-modules/kubernetes-engine/google//modules/beta-autopilot-private-cluster"
  version                      = "28.0.0"
  project_id                   = var.project_id
  name                         = "xx-${var.env}"
  kubernetes_version           = "1.27.3-gke.1700"
  region                       = var.region
  network                      = local.network
  subnetwork                   = local.subnetwork
  master_ipv4_cidr_block       = var.control_plane_ipv4
  authenticator_security_group = "xx"

  ip_range_pods              = "pods"
  ip_range_services          = "services"
  enable_cost_allocation     = true
  enable_private_nodes       = true
  add_cluster_firewall_rules = true
  maintenance_start_time     = "06:00"
  firewall_inbound_ports  = ["8000", "8443", "9443", "15017"]

}
```

### Terraform Version

```sh
Terraform v1.6.0
```

### Additional information

[local.cluster_network_tag definition](https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/blob/master/modules/beta-autopilot-private-cluster/main.tf#L111)
[sample firewall rule](https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/blob/master/modules/beta-autopilot-private-cluster/firewall.tf#L36)
[node_pool_auto_config](https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/blob/master/modules/beta-autopilot-private-cluster/cluster.tf#L100)

**Comment by GorginZ** [comment]
hi @TobiasLierzer  I came across this too. I have a PR open that's being looked at. Any thoughts are welcome! https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/pull/1817
I've only done for the private cluster, but pending feedback we'll want to fix for other autopilot configs too.

**Comment by TobiasLierzer** [comment]
I just checked your PR. Great thought making the tag dependent on the three firewall variables.  

Closing this issue as it is a duplicate of
https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/issues/1230
